clarity/src/vm/clarity_wasm.rs

```diff
@@ -0,0 +1,1035 @@
+}
+
+impl<'a, 'b> ClarityWasmInitContext<'a, 'b> {
+    pub fn new(
+        global_context: &'b mut GlobalContext<'a>,
+        contract_context: &'b mut ContractContext,
+        contract_analysis: &'b ContractAnalysis,
+    ) -> Self {
+        ClarityWasmInitContext {
+            run_context: ClarityWasmRunContext {
+                global_context,
+                contract_context,
+            },
+            contract_analysis,
+        }
+    }
+}
+
+impl ClarityWasmContext for ClarityWasmInitContext<'_, '_> {
+    fn contract_identifier(&self) -> &QualifiedContractIdentifier {
+        &self.run_context.contract_context.contract_identifier
+    }
+
+    fn get_var_metadata(&self, name: &str) -> Option<&DataVariableMetadata> {
+        self.run_context.contract_context.meta_data_var.get(name)
+    }
+
+    fn lookup_variable_with_size(
+        &mut self,
+        contract_identifier: &QualifiedContractIdentifier,
+        variable_name: &str,
+        variable_descriptor: &DataVariableMetadata,
+    ) -> Result<ValueResult, Error> {
+        self.run_context
+            .global_context
+            .database
+            .lookup_variable_with_size(contract_identifier, variable_name, variable_descriptor)
+    }
+
+    fn set_variable(
+        &mut self,
+        contract_identifier: &QualifiedContractIdentifier,
+        variable_name: &str,
+        value: Value,
+        variable_descriptor: &DataVariableMetadata,
+    ) -> Result<ValueResult, Error> {
+        self.run_context.global_context.database.set_variable(
+            contract_identifier,
+            variable_name,
+            value,
+            variable_descriptor,
+        )
+    }
+}
+
+pub fn initialize_contract(
+    global_context: &mut GlobalContext,
+    contract_context: &mut ContractContext,
+    contract_analysis: &ContractAnalysis,
+) -> Result<Option<Value>, Error> {
+    let context = ClarityWasmInitContext::new(global_context, contract_context, contract_analysis);
+    let engine = Engine::default();
+    let module = context
+        .run_context
+        .contract_context
+        .with_wasm_module(|wasm_module| {
+            Module::from_binary(&engine, wasm_module)
+                .map_err(|e| Error::Wasm(WasmError::UnableToLoadModule(e)))
+        })?;
+    let mut store = Store::new(&engine, context);
+    let mut linker = Linker::new(&engine);
+
+    // Link in the host interface functions.
+    link_define_function_fn(&mut linker);
+    link_define_variable_fn(&mut linker);
+    link_get_variable_fn(&mut linker);
+    link_set_variable_fn(&mut linker);
+    link_log(&mut linker);
+
+    let instance = linker.instantiate(store.as_context_mut(), &module).unwrap();
+
+    // Call the `.top-level` function, which contains all top-level expressions
+    // from the contract.
+    let func = instance
+        .get_func(store.as_context_mut(), ".top-level")
+        .expect(".top-level function was not found in the generated WASM binary.");
```

Comment:
Will likely want proper error handling instead of panic

Reply:
4565abf70 replaces all panics with errors.

Reply:
Please re-review to make sure I didn't miss any.